packages/backend/src/controllers/people/people.controller.test.ts

```diff
@@ -66,4 +72,135 @@ describe('People Controller', () => {
+        name: 'Luke Skywalker',
+        mass: 77,
+        height: 172,
+        skinColor: 'black',
+      });
+    });
+
+    it('should throw 404 if mass is negative', async () => {
+      const { ctx, next } = createContextMock();
+
+      ctx.request.body = { ...person, mass: -10 };
+
+      await addPerson(ctx, next);
+
+      expect(ctx.status).toBe(StatusCodes.BAD_REQUEST);
+      expect(ctx.body).toBe('Height or mass cannot be a negative number.');
+    });
+
+    it('should throw 404 if height is negative', async () => {
+      const { ctx, next } = createContextMock();
+
+      ctx.request.body = { ...person, height: -10 };
+
+      await addPerson(ctx, next);
+
+      expect(ctx.status).toBe(StatusCodes.BAD_REQUEST);
+      expect(ctx.body).toBe('Height or mass cannot be a negative number.');
+    });
+  });
+
+  describe('patchPerson', () => {
+    const person = {
+      name: 'Luke Skywalker',
+      mass: 77,
+      height: 172,
+      skinColor: 'black',
+    };
+
+    it('should update and return person', async () => {
+      const { ctx, next } = createContextMock();
+      prismaMock.person.update.mockResolvedValue(person as Person);
+
+      ctx.params = {
+        id: '1',
+      };
+      ctx.request.body = person;
+
+      await patchPerson(ctx, next);
+
+      expect(next).toBeCalled();
+      expect(ctx.status).toBe(StatusCodes.OK);
+      expect(ctx.body).toStrictEqual({
+        name: 'Luke Skywalker',
+        mass: 77,
+        height: 172,
+        skinColor: 'black',
+      });
+    });
+
+    it('should throw 404 if mass is negative', async () => {
+      const { ctx, next } = createContextMock();
+
+      ctx.params = {
+        id: '1',
+      };
+      ctx.request.body = { ...person, mass: -10 };
+
+      await patchPerson(ctx, next);
+
+      expect(ctx.status).toBe(StatusCodes.BAD_REQUEST);
+      expect(ctx.body).toBe('Height or mass cannot be a negative number.');
+    });
+
+    it('should throw 404 if height is negative', async () => {
+      const { ctx, next } = createContextMock();
+
+      ctx.params = {
+        id: '1',
+      };
+      ctx.request.body = { ...person, height: -10 };
+
+      await patchPerson(ctx, next);
+
+      expect(ctx.status).toBe(StatusCodes.BAD_REQUEST);
+      expect(ctx.body).toBe('Height or mass cannot be a negative number.');
+    });
+  });
+
+  describe('removePerson', () => {
+    it('should remove person', async () => {
+      const { ctx, next } = createContextMock();
+      prismaMock.person.delete.mockResolvedValue({} as Person);
+
+      ctx.params = {
+        id: '1',
+      };
+
+      await removePerson(ctx, next);
+
+      expect(next).toBeCalled();
+      expect(ctx.status).toBe(StatusCodes.NO_CONTENT);
+    });
+
+    it('should throw 404 - person not found', async () => {
+      const { ctx, next } = createContextMock();
+
+      await removePerson(ctx, next);
+
+      expect(ctx.status).toBe(StatusCodes.NOT_FOUND);
```

Comment:
Nie sprawdzasz tu czy funkcja `removePerson` zwraca błąd, tylko czy status jest typu NOT_FOUND.

Do sprawdzania tego czy funkcja się wykrzaczyła służy funkcja: https://jestjs.io/docs/expect#tothrowerror

Reply:
Tak - poprawiłem to. Powinno być "should return 400 - person not found"! :-)